dspace-api/src/main/java/org/dspace/external/provider/impl/LiveImportDataProvider.java

```diff
@@ -0,0 +1,162 @@
+     */
+    public void setRecordIdMetadata(String recordIdMetadata) {
+        this.recordIdMetadata = recordIdMetadata;
+    }
+
+    /**
+     * This method set the dublin core identifier to display the title
+     * @param displayMetadata metadata to use as title
+     */
+    public void setDisplayMetadata(String displayMetadata) {
+        this.displayMetadata = displayMetadata;
+    }
+
+    @Override
+    public Optional<ExternalDataObject> getExternalDataObject(String id) {
+        try {
+            ExternalDataObject externalDataObject = getExternalDataObject(querySource.getRecord(id));
+            return Optional.of(externalDataObject);
+        } catch (MetadataSourceException e) {
+            throw new RuntimeException(
+                    "The live import provider " + querySource.getImportSource() + " throws an exception", e);
+        }
+    }
+
+    @Override
+    public List<ExternalDataObject> searchExternalDataObjects(String query, int start, int limit) {
+        Collection<ImportRecord> records;
+        try {
+            records = querySource.getRecords(query, start, limit);
+            return records.stream().map(r -> getExternalDataObject(r)).collect(Collectors.toList());
+        } catch (MetadataSourceException e) {
+            throw new RuntimeException(
+                    "The live import provider " + querySource.getImportSource() + " throws an exception", e);
+        }
+    }
+
+    @Override
+    public boolean supports(String source) {
+        return StringUtils.equalsIgnoreCase(sourceIdentifier, source);
+    }
+
+    @Override
+    public int getNumberOfResults(String query) {
+        try {
+            return querySource.getRecordsCount(query);
+        } catch (MetadataSourceException e) {
+            throw new RuntimeException(
+                    "The live import provider " + querySource.getImportSource() + " throws an exception", e);
+        }
+    }
+
+    /**
+     * Internal method to convert an ImportRecord to an ExternalDataObject
+     * 
+     * FIXME it would be useful to remove ImportRecord at all in favor of the
+     * ExternalDataObject
+     * 
+     * @param record
+     * @return
+     */
+    private ExternalDataObject getExternalDataObject(ImportRecord record) {
+        //return 400 if no record were found
+        if (record == null) {
+            throw new IllegalArgumentException("No record found for query or id");
+        }
+        ExternalDataObject externalDataObject = new ExternalDataObject(sourceIdentifier);
+        String id = getFirstValue(record, recordIdMetadata);
+        String display = getFirstValue(record, displayMetadata);
+        externalDataObject.setId(id);
+        externalDataObject.setDisplayValue(display);
+        externalDataObject.setValue(display);
+        for (MetadatumDTO dto : record.getValueList()) {
+            // FIXME it would be useful to remove MetadatumDTO in favor of MetadataValueDTO
```

Comment:
similar to https://github.com/DSpace/DSpace/pull/2824/files#r477205482 this for convert MetadatumDTO into MetadataValueDTO but it would be better if we only have MetadataValueDTO at all